File: frontend/src/objects/rider/RiderDeliverySettlement.ts
import type { OrderId } from '@/objects/shared/ids'

export interface RiderDeliverySettlement {
  ok: boolean
  orderId: OrderId
  earnedEnergy: number
  currentEnergyPoints: number
  currentTimeoutCardCount: number
  wasTimeout: boolean
  timeoutCardUsed: boolean
  timeoutExempted: boolean
  overtimeSeconds: number
}

File: backend/src/rider/objects/RiderDeliverySettlement.scala
package delivery.rider.objects

import delivery.domain.OrderId

final case class RiderDeliverySettlement(
    ok: Boolean,
    orderId: OrderId,
    earnedEnergy: Int,
    currentEnergyPoints: Int,
    currentTimeoutCardCount: Int,
    wasTimeout: Boolean,
    timeoutCardUsed: Boolean,
    timeoutExempted: Boolean,
    overtimeSeconds: Int
)
